### PR TITLE
[panorama] force pyramid levels count in compositing

### PR DESCRIPTION
**Pull request by servantftechnicolor**
[panorama] force pyramid levels count in compositing

### CODE REVIEW

File: src/software/pipeline/main_panoramaCompositing.cpp [hunk 1]
```diff
@@ -620,15 +629,16 @@ int aliceVision_main(int argc, char** argv)
...
-        "overlayType,c", po::value<std::string>(&overlayType)->required(), "Overlay Type [none, borders, seams, all].")(
-        "storageDataType", po::value<image::EStorageDataType>(&storageDataType)->default_value(storageDataType),
-        ("Storage data type: " + image::EStorageDataType_informations()).c_str())(
-        "rangeIteration", po::value<int>(&rangeIteration)->default_value(rangeIteration),
-        "Range chunk id.")("rangeSize", po::value<int>(&rangeSize)->default_value(rangeSize), "Range size.")(
-        "maxThreads", po::value<int>(&maxThreads)->default_value(maxThreads), "max number of threads to use.")(
-        "labels,l", po::value<std::string>(&labelsFilepath)->required(), "Labels image from seams estimation.");
+    optionalParams.add_options()
+        ("compositerType,c", po::value<std::string>(&compositerType)->required(), "Compositer Type [replace, alpha, multiband].")
+        ("forcedLevelsCount, f", po::value<int>(&forcedLevelsCount)->default_value(forcedLevelsCount), "For multiband compositer, force a given number of levels in pyramid (0 for automatic)")
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaCompositing.cpp:634]:
```suggestion
        ("forceMinLevels,f", po::value<int>(&forceMinLevels)->default_value(forceMinLevels), "For multiband compositer, force a minimum number of levels in the image pyramid (0 for automatic)")
```

**Review by fabiencastan — CHANGES_REQUESTED** [review]
It would be better if the option defines a min pyramid level value, instead of overriding the automatically determined number of pyramid levels.

File: src/software/pipeline/main_panoramaCompositing.cpp (hunk 1 above)
Comment (servantftechnicolor) [reply src/software/pipeline/main_panoramaCompositing.cpp]: done

File: src/software/pipeline/main_panoramaCompositing.cpp [hunk 2]
```diff
@@ -115,6 +115,14 @@ std::unique_ptr<PanoramaMap> buildMap(const sfmData::SfMData& sfmData, const std
...
     }
 
+    ALICEVISION_LOG_INFO("Estimated levels count : " << min_scale);
+
+    if (forceMinPyramidLevels > 0)
+    {
+        min_scale = std::max(forceMinPyramidLevels, min_scale);
+        ALICEVISION_LOG_INFO("Forced levels count : " << min_scale);
+    }
+
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaCompositing.cpp:125]:
```suggestion
    ALICEVISION_LOG_INFO("Estimated pyramid levels count: " << min_scale);

    if (forceMinPyramidLevels > min_scale)
    {
        min_scale = forceMinPyramidLevels;
        ALICEVISION_LOG_INFO("Forced pyramid levels count: " << min_scale);
    }

```

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**